test/state/HypermediaState.test.js

```diff
@@ -435,6 +435,34 @@ describe('HypermediaState class', () => {
 			});
 
 		});
+
+	});
+
+	it('entity is a subentity of itself', async() => {
```

Comment:
Remember to title tests as what the behaviour is. In this case,
```suggestion
	it('pushes each state only once when there is a circular dependency', async() => {
```